master/master.cfg

```diff
@@ -209,6 +211,90 @@ def add_get_source_steps(factory, llvm_branch):
                         branch=llvm_branch,
                         mode='incremental'))
 
+# Determined by running `set` in cmd.exe before and after vcvarsall.bat
+# and diffing the output. It's likely that we don't need all of these
+# to make things work, but I haven't bothered to figure out what is irrelevant,
+# so I'm erring on the side of maybe too much.
+VCVARSALL_ENV_VARS = [
+    "COMMANDPROMPTTYPE",
+    "DEVENVDIR",
+    "EXTENSIONSDKDIR",
+    "FRAMEWORK40VERSION",
+    "FRAMEWORKDIR",
+    "FRAMEWORKDIR64",
+    "FRAMEWORKVERSION",
+    "FRAMEWORKVERSION64",
+    "INCLUDE",
+    "LIB",
+    "LIBPATH",
+    "NETFXSDKDIR",
+    "PATH",
+    "PLATFORM",
+    "UCRTVERSION",
+    "UNIVERSALCRTSDKDIR",
+    "VCIDEINSTALLDIR",
+    "VCINSTALLDIR",
+    "VCTOOLSINSTALLDIR",
+    "VCTOOLSREDISTDIR",
+    "VCTOOLSVERSION",
+    "VISUALSTUDIOVERSION",
+    "VS110COMNTOOLS",
+    "VS120COMNTOOLS",
+    "VS140COMNTOOLS",
+    "VS160COMNTOOLS",
+    "VSCMD_ARG_APP_PLAT",
+    "VSCMD_ARG_HOST_ARCH",
+    "VSCMD_ARG_TGT_ARCH",
+    "VSCMD_VER",
+    "VSINSTALLDIR",
+    "WINDOWSLIBPATH",
+    "WINDOWSSDKBINPATH",
+    "WINDOWSSDKDIR",
+    "WINDOWSSDKLIBVERSION",
+    "WINDOWSSDKVERBINPATH",
+    "WINDOWSSDKVERSION",
+    "WINDOWSSDK_EXECUTABLEPATH_X64",
+    "WINDOWSSDK_EXECUTABLEPATH_X86",
+    "__DOTNET_ADD_64BIT",
+    "__DOTNET_PREFERRED_BITNESS",
+    "__VSCMD_PREINIT_PATH",
+    "__VSCMD_SCRIPT_ERR_COUNT",
+]
+
+
+def get_msvc_config_steps(factory, os):
+    if os.startswith('win'):
+        if '-32' in os:
+            arch = ' x86'
```

Comment:
We might want this to be `x86_amd64` so we don't run out of memory in the linking step. I've found this is necessary for LLVM, but not (yet) for Halide.

Reply:
Should be `amd64_x86`, oops.

Reply:
To make sure I understand: This means it uses the 64-bit toolchain to produce 32-bit code (so using >2GB of ram is no big deal)?

Reply:
Yeah, nomenclature is confusing: https://docs.microsoft.com/en-us/cpp/build/building-on-the-command-line?view=vs-2019